### PR TITLE
chore(deps): update some rust crypto dependencies

### PR DESCRIPTION
**Pull request by rjzak**
Attempting to update some Rust Crypto dependencies separately since https://github.com/enarx/enarx/pull/2560 didn't go so well.

### CODE REVIEW

**Comment by rjzak** [comment]
Closing & re-opening to try to trigger CI tests to run.

**Comment by rjzak** [comment]
Some errors are because the integration tests try to run the `nil` backend with an executable, but that's not allowed: https://github.com/enarx/enarx/blob/b999fd1f7dc1824e621d509eefed7ff2069f15fb/src/backend/nil.rs#L57-L59

**Comment by haraldh** [comment]
For the network error I filed this: https://github.com/bytecodealliance/wasmtime/pull/7686

**Comment by haraldh** [comment]
additional networking test fixes: https://github.com/enarx/enarx/pull/2562

**Comment by haraldh** [comment]
disable exec tests for nil backend: https://github.com/enarx/enarx/pull/2563

**Comment by rjzak** [comment]
@haraldh I merged your PRs and rebased. What do you think of this PR?

**Comment by rjzak** [comment]
After this, I'll update other deps then update the Rust version (can't do that now since some newer deps depend on newer Rust).

**Review by haraldh — APPROVED** [review]

File: crates/exec-wasmtime/src/runtime/identity/mod.rs [hunk 1]
```diff
@@ -74,28 +79,27 @@ fn csr(pki: &PrivateKeyInfo<'_>, exts: Vec<Extension<'_>>) -> anyhow::Result<Vec
...
-    .to_vec()
+    .to_der()
     .context("failed to encode CSR")
 }
 
 /// Generates a new private key and corresponding CSR
 #[instrument]
 pub fn generate() -> anyhow::Result<(Zeroizing<Vec<u8>>, Vec<u8>)> {
-    let platform = Platform::get().context("failed to query platform")?;
+    let platform = Platform::get().unwrap(); //.context("failed to query platform")?;
```
Comment (haraldh) [inline crates/exec-wasmtime/src/runtime/identity/mod.rs:92]: why?

File: crates/exec-wasmtime/src/runtime/identity/mod.rs [hunk 2]
```diff
@@ -109,17 +113,17 @@ pub fn generate() -> anyhow::Result<(Zeroizing<Vec<u8>>, Vec<u8>)> {
         }
     };
 
-    let attestation_report = platform.attest(&key_hash).context("failed to attest")?;
+    let attestation_report = platform.attest(&key_hash).unwrap(); //.context("failed to attest")?;
```
Comment (haraldh) [inline crates/exec-wasmtime/src/runtime/identity/mod.rs:116]: why?

File: crates/exec-wasmtime/src/runtime/identity/mod.rs [hunk 3]
```diff
@@ -109,17 +113,17 @@ pub fn generate() -> anyhow::Result<(Zeroizing<Vec<u8>>, Vec<u8>)> {
...
 
     // Create extensions.
     let ext = vec![Extension {
         extn_id: platform.technology().into(),
         critical: false,
-        extn_value: &attestation_report,
+        extn_value: OctetString::new(attestation_report).unwrap(),
```
Comment (haraldh) [inline crates/exec-wasmtime/src/runtime/identity/mod.rs:122]: use `context()?` please

File: crates/exec-wasmtime/src/runtime/identity/mod.rs [hunk 4]
```diff
@@ -109,17 +113,17 @@ pub fn generate() -> anyhow::Result<(Zeroizing<Vec<u8>>, Vec<u8>)> {
...
     }];
 
     // Make a certificate signing request.
-    let req = csr(&pki, ext).context("failed to generate a CSR")?;
+    let req = csr(&pki, ext).unwrap(); //.context("failed to generate a CSR")?;
```
Comment (haraldh) [inline crates/exec-wasmtime/src/runtime/identity/mod.rs:126]: why?

**Review by haraldh — CHANGES_REQUESTED** [review]

File: crates/exec-wasmtime/src/runtime/identity/mod.rs (hunk 1 above)
Comment (rjzak) [reply crates/exec-wasmtime/src/runtime/identity/mod.rs]: Sorry, this was left over from me debugging some crypto issues.

File: crates/exec-wasmtime/src/runtime/identity/mod.rs (hunk 2 above)
Comment (rjzak) [reply crates/exec-wasmtime/src/runtime/identity/mod.rs]: Sorry, leftover debugging cruft

File: crates/exec-wasmtime/src/runtime/identity/mod.rs (hunk 3 above)
Comment (rjzak) [reply crates/exec-wasmtime/src/runtime/identity/mod.rs]: Fixed

File: crates/exec-wasmtime/src/runtime/identity/mod.rs (hunk 4 above)
Comment (rjzak) [reply crates/exec-wasmtime/src/runtime/identity/mod.rs]: Fixed

**Pull request merged**